fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryTextIndexPlan.java

```diff
@@ -162,4 +168,40 @@ public int planHash() {
     public String toString() {
         return "TextIndex(" + textScan.getIndex().getName() + " " + textScan.getGroupingComparisons() + ", " + textScan.getTextComparison() + ", " + textScan.getSuffixComparisons() + ")";
     }
+
+    /**
+     * Rewrite the planner graph for better visualization of a query index plan.
+     * @param childGraphs planner graphs of children expression that already have been computed
+     * @return the rewritten planner graph that models the index as a separate node that is connected to the
+     *         actual index scan plan node.
+     */
+    @Nonnull
+    @Override
+    public PlannerGraph rewritePlannerGraph(@Nonnull final List<? extends PlannerGraph> childGraphs) {
+        final ImmutableMap.Builder<String, Attribute> additionalAttributesBuilder = ImmutableMap.builder();
+        if (textScan.getGroupingComparisons() != null) {
+            additionalAttributesBuilder.put("groupingComparisons", Attribute.gml(Objects.requireNonNull(textScan.getGroupingComparisons()).toString()));
+        } else {
+            additionalAttributesBuilder.put("groupingComparisons", Attribute.gml("none"));
+        }
+        if (textScan.getSuffixComparisons() != null) {
+            additionalAttributesBuilder.put("suffixComparisons", Attribute.gml(Objects.requireNonNull(textScan.getSuffixComparisons()).toString()));
+        } else {
+            additionalAttributesBuilder.put("suffixComparisons", Attribute.gml("none"));
+        }
+        additionalAttributesBuilder.put("textComparisons", Attribute.gml(Objects.requireNonNull(textScan.getTextComparison()).toString()));
+
+        final PlannerGraph.Node root =
+                new PlannerGraph.OperatorNodeWithInfo(this,
+                        NodeInfo.TEXT_INDEX_SCAN_OPERATOR,
+                        ImmutableList.of("grouping comparisons: {{groupingComparisons}}",
+                                "text comparisons: {{textComparisons}}",
+                                "suffix comparisons: {{suffixComparisons}}"),
```

Comment:
This is really messy but I think it's the best we can do with the current text scan plan. This will be much better someday :-)